packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -1,89 +1,121 @@
 import { FormPoster } from '@bigcommerce/form-poster';
 
 import { CheckoutActionCreator, CheckoutStore } from '../../../checkout';
-import { InvalidArgumentError, MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
-import { ApproveDataOptions, ButtonsOptions, ClickDataOptions, PaypalCommercePaymentProcessor } from '../../../payment/strategies/paypal-commerce';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType, } from '../../../common/error/errors';
+import { PaymentMethodClientUnavailableError } from '../../../payment/errors';
+import { ApproveDataOptions, ButtonsOptions, PaypalButtonStyleOptions, PaypalCommerceRequestSender, PaypalCommerceScriptLoader, PaypalCommerceSDK } from '../../../payment/strategies/paypal-commerce';
 import { CheckoutButtonInitializeOptions } from '../../checkout-button-options';
 import CheckoutButtonStrategy from '../checkout-button-strategy';
 
+import getValidButtonStyle from './get-valid-button-style';
+
 export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrategy {
-    private _isCredit?: boolean;
-    private _isVenmo?: boolean;
-    private _isVenmoEnabled?: boolean;
+    private _paypalCommerceSdk?: PaypalCommerceSDK;
 
     constructor(
         private _store: CheckoutStore,
         private _checkoutActionCreator: CheckoutActionCreator,
         private _formPoster: FormPoster,
-        private _paypalCommercePaymentProcessor: PaypalCommercePaymentProcessor
+        private _paypalScriptLoader: PaypalCommerceScriptLoader,
+        private _paypalCommerceRequestSender: PaypalCommerceRequestSender
     ) {}
 
     async initialize(options: CheckoutButtonInitializeOptions): Promise<void> {
-        let state = this._store.getState();
-        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(options.methodId);
-        const { initializationData } = paymentMethod;
+        const { paypalcommerce, containerId, methodId } = options;
+        const { initializesOnCheckoutPage, style } = paypalcommerce || {};
```

Comment:
Do we need the empty object since https://github.com/bigcommerce/checkout-sdk-js/pull/1570/files#diff-a0e269a19247a2fe2de4543b3db3dc1cc7b26aeb62b056f0c6ecab82d449683cR35 we do validate?

Reply:
We can avoid using empty object by moving this line under the `paypalcommerce` object check. But if we want to left this line on it's place, the typescript will have an error.